packages/myst-transforms/src/links/doi.ts

```diff
@@ -30,7 +30,7 @@ export class DOITransformer implements LinkTransformer {
       return false;
     }
     link.url = doiUrl;
-    link.data = { doi: doi.normalize(doiUrl) };
+    link.data = { ...link.data, doi: doi.normalize(doiUrl) };
```

Comment:
This was the change that fixed my immediate upstream issue, and has been tested. The `doi` field is added alongside other userdata.